rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -111,28 +125,30 @@ case class PrettyPrinter(freeShift: Int, boundShift: Int) {
               else ""
             })
         }
-        "for( " + bindsString + " ) { " + PrettyPrinter(freeShift, boundShift + totalFree)
+
+        "for( " + bindsString + " ) { " + this
+          .copy(boundShift = boundShift + totalFree)
           .buildString(r.body.get) + " }"
 
       case e: Eval => "*" + buildString(e.channel.get)
 
       case n: New =>
-        "new " + buildVariables(n.bindCount) + " in { " + PrettyPrinter(
-          freeShift,
-          boundShift + n.bindCount).buildString(n.p.get) + " }"
+        "new " + buildVariables(n.bindCount) + " in { " + this
+          .copy(boundShift = boundShift + n.bindCount)
+          .buildString(n.p.get) + " }"
 
       case e: Expr =>
         buildString(e)
 
       case m: Match =>
-        "match { " + buildString(m.target.get) + " } { " + {
-          for { (matchCase, i) <- m.cases.zipWithIndex } yield {
-            buildMatchCase(matchCase) + {
-              if (i != m.cases.length - 1) "; "
-              else ""
-            }
-          }
-        } + " }"
+        "match " + buildString(m.target.get) + " { " +
+          ("" /: m.cases.zipWithIndex) {
+            case (string, (matchCase, i)) =>
+              string + buildMatchCase(matchCase) + {
+                if (i != m.cases.length - 1) " ; "
```

Comment:
Don't fix this now, because it was there before, but there's no `;` separating cases.